### PR TITLE
Add Editor Area Onboarding for Java Perspective

### PR DESCRIPTION
**Pull request by BeckerWdf**
Fixes: #505

### CODE REVIEW

**Comment by vogella** [comment]
Awesome. Thanks @BeckerWdf 

@iloveeclipse please review

File: org.eclipse.jdt.ui/plugin.xml [hunk 1]
```diff
@@ -610,10 +610,25 @@
          <showInPart id="org.eclipse.jdt.ui.PackageExplorer"/>
       </perspectiveExtension>
 
-      <perspectiveExtension targetID="org.eclipse.jdt.ui.JavaPerspective">
+      <perspectiveExtension
+            editorOnboardingImage="$nl$/icons/full/onboarding_jperspective.png"
+            editorOnboardingText="perspectiveExtension.editorOnboardingText"
```
Comment (iloveeclipse) [inline org.eclipse.jdt.ui/plugin.xml:615]: The "%" is missing

File: org.eclipse.jdt.ui/plugin.xml [hunk 2]
```diff
@@ -610,10 +610,25 @@
...
+         </editorOnboardingCommand>
+         <editorOnboardingCommand
+               id="org.eclipse.ui.newWizard">
+         </editorOnboardingCommand>
+         <editorOnboardingCommand
+               id="org.eclipse.jdt.ui.navigate.open.type">
+         </editorOnboardingCommand>         
       </perspectiveExtension>
 
       <perspectiveExtension targetID="org.eclipse.jdt.ui.JavaBrowsingPerspective">
```
Comment (iloveeclipse) [inline org.eclipse.jdt.ui/plugin.xml:634]: Two perspectives below would also profit from same addition.

**Review by iloveeclipse — CHANGES_REQUESTED** [review]

File: org.eclipse.jdt.ui/plugin.xml (hunk 1 above)
Comment (BeckerWdf) [reply org.eclipse.jdt.ui/plugin.xml]: Good catch. I fixed it.

File: org.eclipse.jdt.ui/plugin.xml (hunk 2 above)
Comment (BeckerWdf) [reply org.eclipse.jdt.ui/plugin.xml:634]: I need to create the icons first. This will take some more time.

**Comment by BeckerWdf** [comment]
Can someone on windows pls. also check if the icon also looks good on windows so that we don't run into the same issue as in https://github.com/eclipse-platform/eclipse.platform.ui/issues/687

File: org.eclipse.jdt.ui/plugin.xml (hunk 2 above)
Comment (BeckerWdf) [reply org.eclipse.jdt.ui/plugin.xml:634]: The different perspectives have different icons and so should the onboarding info.

**Review by iloveeclipse — APPROVED** [review]
LGTM.

Here is how it looks like on Linux / no themes:

image

**Comment by iloveeclipse** [comment]
> Can someone on windows pls. also check if the icon also looks good on windows so that we don't run into the same issue as in [eclipse-platform/eclipse.platform.ui#687](https://github.com/eclipse-platform/eclipse.platform.ui/issues/687)

Looks almost same as on Linux (on Widows "Light" theme, with and without Eclipse themes).

File: org.eclipse.jdt.ui/plugin.xml (hunk 2 above)
Comment (BeckerWdf) [reply org.eclipse.jdt.ui/plugin.xml]: > Two perspectives below would also profit from same addition.

This is done now.

**Review by iloveeclipse — APPROVED** [review]
Thanks a lot. Looks good.

**Comment by BeckerWdf** [comment]
> Thanks a lot. Looks good.

Can you post screenshots how it look in light and dark?

**Comment by iloveeclipse** [comment]
> Can you post screenshots how it look in light and dark?

I don't use dark theme and won't mess up with my Gnome config to set it in the system.
But I will post the "light" version.

**Comment by iloveeclipse** [comment]
image
image
image

**Comment by BeckerWdf** [comment]
> I don't use dark theme and won't mess up with my Gnome config to set it in the system.

Ok. No problem.

> But I will post the "light" version.

They look nice

**Comment by BeckerWdf** [comment]
From my point of view this can be merged.
Can somebody with jdt-ui committer rights merge?

**Comment by vogella** [comment]
The inner white filling looks bad in the dark theme. @BeckerWdf could you do the same as in the Resource perspective and remove it?

image

**Comment by BeckerWdf** [comment]
> you do the same as in the Resource perspective and remove it?

I am not sure. Because then it would not longer be a "solid" ball but only a circle with not fill - especially in the dark theme. Maybe I can play wit transparency and put in a fill which is almost transparent. By doing this. You would also have a fill in the dark theme but a much darker one.

**Comment by BeckerWdf** [comment]
> The inner white filling looks bad in the dark theme. @BeckerWdf could you do the same as in the Resource perspective and remove it?
> 
> image

See https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/516

**Pull request merged**